File: ShadowFrameworkLite20/src/shadow/image/SFBufferData.java
package shadow.image;

/** 
 * Describe a buffer
 * 
 * @author Alessandro
 */
public abstract class SFBufferData {

	private int width,height;
	private SFImageFormat format;
	
	public SFBufferData(int width, int height, SFImageFormat format) {
		super();
		this.width = width;
		this.height = height;
		this.format = format;
	}

	public int getWidth() {
		return width;
	}

	public int getHeight() {
		return height;
	}

	public SFImageFormat getFormat() {
		return format;
	}
	
}

File: trunk/SFDataApplicationTestC++/shadow/image/SFBufferData.cpp

#include "SFBufferData.h"


namespace sf{

SFBufferData::SFBufferData(int width, int height, SFImageFormat format) {
    this->width = width;
    this->height = height;
    this->format = format;
}

int SFBufferData::getWidth() {
    return width;
}

int SFBufferData::getHeight() {
    return height;
}

SFImageFormat SFBufferData::getFormat() {
    return format;
}

}
